File: Solutions/0001.py
class Solution(object):
  def twoSum(self, nums, target):
    numMap = {}
    for i, num in enumerate(nums):
        numMap[num] = i
    for i, num in enumerate(nums):
        complement = target - num
        if complement in numMap.keys() and numMap[complement] != i:
            return [i, numMap[complement]]
    return []

File: Solutions_in_java/0001.java
    class Solution {
        public int[] twoSum(int[] nums, int target) {
            // int[] res = new int[2];
            // number, ind hashmap
            HashMap<Integer, Integer> hm = new HashMap<>();
            for(int i = 0; i < nums.length; i++){
                if(hm.containsKey(target - nums[i])){
                    // res = new int[] {hm.get(target - nums[i]), i};
                    return new int[] {hm.get(target - nums[i]), i};
                }
                hm.put(nums[i], i);
            }
            // return res;
            return new int[] {};
        }
    }